pkg/sources/gitlab/gitlab.go

```diff
@@ -47,6 +47,8 @@ type Source struct {
 	git                    *git.Git
 	scanOptions            *git.ScanOptions
 
+	apiClient *gitlab.Client
```

Comment:
it looks like you've added this here so that you don't have to pass a client argument into `scanTargets` and its call tree. but now there's one path through the code that takes the client via parameter (the full-scan path - including units support) and one that uses the struct field (the targeted scan). i find this unnecessarily difficult to track mentally, and it feels like an accident waiting to happen, so i think we should settle on just doing it one way, at _least_ within a single source. (we should probably standardize _across_ sources eventually, but that's a bigger project than this pr.)

both approaches have upsides and downsides: the parameter method is easier to reason about but requires initializing the client more often; the field method means that we can theoretically initialize the client only once, but have to manage its lifetime. (which, as your other pr highlights, is a non-trivial problem!)

Reply:
I'll pass the client. 👍